script.js

```diff
@@ -0,0 +1,54 @@
+const svelte = require("svelte/compiler");
+const autoprocessor = require("svelte-preprocess");
+const fs = require("fs");
+const path = require("path");
+const glob = require("glob");
+
+const runCommand = async () => {
+  // source file paths
+  const srcPath = path.join(__dirname, "src");
+
+  // read glob of files in directory
+  glob(path.join(srcPath, "**/*.svelte"), {}, (error, files) => {
+    // handling error
+    if (error) {
+      return console.log("Unable to scan directory: " + error);
+    }
+    // listing all files using forEach
+    files.forEach(async (file) => {
+      // load file
+      const sourceFile = fs.readFileSync(file, "utf-8");
+      const distFile = file.replace("/src/", "/dist/");
+
+      // process .svelte file
+      if (file.includes(".svelte")) {
+        // run autopreprocessor
+        await ParseSvelte(sourceFile, distFile);
+      }
+      // copy other files
+      else {
+        // copy static files
+        fs.copyFileSync(file, distFile);
```

Comment:
Just copying other files will not work if they need preprocessing, too, like TS files.

Reply:
The svelte pre-processor can handle the .svelte files, i was thinking of using the typescript plugin for rollup to compile the .ts files after the preprocessing. I can just get rid of the copy file logic to prevent the .ts files from going into the output folder.

as an alternative, i could change the postbuild to a prebuild script so that the preprocessor can spit our compiled .svelte files and then have rollup bundle/minify the remaining .svelte and .ts files.

Reply:
Note typescript (and maybe other) compiler can simply be called directly on the source with the output directory specified. 
`tsc --declaration --outDir dist`
It will then compile the files over to the out directory. Done this way, I would say copying the files over is unnecessary. Perhaps we should *only* worry about svelte files in this script, and let any non-svelte files be handled separately (possibly by their native compilers)